packages/server/src/fhir/operations/expand.ts

```diff
@@ -244,9 +252,10 @@ async function computeExpansion(
     const codeSystem = codeSystemCache[include.system] ?? (await findTerminologyResource('CodeSystem', include.system));
     codeSystemCache[include.system] = codeSystem;
     if (include.concept) {
-      const concepts = await Promise.all(include.concept.flatMap((c) => validateCoding(codeSystem, c)));
-      for (const c of concepts) {
-        if (c && (!filter || c.display?.includes(filter))) {
+      const filteredCodings = filterCodings(include.concept, filter);
+      const validCodings = await Promise.all(filteredCodings.map((c) => validateCoding(codeSystem, c)));
```

Comment:
👍 

Aside: we should really prioritize optimizing `validateCoding()` - low hanging fruit here.